### PR TITLE
Update _index.md

### PR DESCRIPTION
**Pull request by cmo-pomerium**
**What is this feature?**

Updating the docs with links to self-hosted open-source reverse proxies for users, including guides (if applicable) to smooth out user journey.

**Why do we need this feature?**

Not applicable

**Who is this feature for?**

Users. They want to secure their Grafana instances, and it just says "Require all network requests being made by Grafana to go through a proxy server."

**Which issue(s) does this PR fix?**:
Not applicable

**Special notes for your reviewer:**

I debated with myself if this was where it belonged, and decided yes. The section seems a bit bare without a few options for users.

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by chri2547** [comment]
@cmo-pomerium , thank you for the contribution. you'll need to run prettier before we can review the PR. Thank you.

**Comment by chri2547** [comment]
@linoman , @IevaVasiljeva - can one of you please review this PR. Thank you!

File: docs/sources/setup-grafana/configure-security/_index.md [hunk 1]
```diff
@@ -39,6 +39,12 @@ There are many firewall tools available, refer to the documentation for your spe
 
 Require all network requests being made by Grafana to go through a proxy server.
 
+Self-hosted reverse proxy options include:
```
Comment (IevaVasiljeva) [inline docs/sources/setup-grafana/configure-security/_index.md:42]: @chri2547 what do you think about this suggestion? I just want to make it clear that these are some examples, but we don't in any way prefer them over other proxy options.

```suggestion
Self-hosted reverse proxy options include but are not limited to:
```

**Review by IevaVasiljeva — APPROVED** [review]
Thanks for your contribution! Overall this looks good to me, just added one small suggestion to make it super clear that this is not an exhaustive list of options.

**Review by linoman — APPROVED** [review]
I think this is ready to be merged. The checks have been failing for all PRs

**Review by chri2547 — APPROVED** [review]

**Pull request merged**